### PR TITLE
update releasing guide.

### PR DESCRIPTION
**Pull request by jlewi**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4698)

### CODE REVIEW

File: docs_dev/releasing.md [hunk 1]
```diff
@@ -150,6 +150,20 @@ A KfDef config file is used to pin the version for the repos we use.
...
+
+On the **master** branch
+
+1. Define a kustomize overlay: `source/vX.Y.Z`
+1. For every kfdef add a patch that overrides the repo spec and version in the KFDef
+   * Look at the existing config
+1. Run `hack/build_kfdef_spec.py` to generate KFDef YAMLs in `kfdefs/`
+1. Check in all modified files
+
+On the **vX.Y.Z-branch** branch
```
Comment (richardsliu) [inline docs_dev/releasing.md]: Currently our branches are organized as vX.Y-branch. Patches do not have their own branches.

**Comment by jlewi** [comment]
Thanks @richardsliu

**Pull request merged**